testbot_description/src/joint_state_publisher.cpp

```diff
@@ -0,0 +1,187 @@
+#include <ros/ros.h>
+#include <string>
+#include <std_msgs/Header.h>
+#include <sensor_msgs/JointState.h>
+#include <tf/transform_broadcaster.h>
+#include <geometry_msgs/TransformStamped.h>
+
+int main(int argc, char **argv){
+
+    ros::init(argc, argv, "joint_state_pub");
+    ros::NodeHandle n;
+
+    ros::Publisher joint_pub = n.advertise<sensor_msgs::JointState>("joint_states", 1);
+    tf::TransformBroadcaster broadcaster;
+    ros::Rate loop_rate(30);
+
+    // declarations
+    const double degree = M_PI/180;
+    double angle = 0;
+
+    //robot state
+    double zero_pos = 0, zero_vel = 0, zero_effort = 0,
+    inc = degree, var = 0,
+    speed = 1, wrench = 0, wrench_inc = 1;
+
+    // message declarations
+    sensor_msgs::JointState joint_state;
+    geometry_msgs::TransformStamped going_around;
+    going_around.header.frame_id = "ground";
+    going_around.child_frame_id = "Pelvis";
+
+    while (ros::ok()){
+
+        //update joint_state
+        joint_state.header.stamp = ros::Time::now();
+        joint_state.name.resize(48);
+        joint_state.name[0] ="jL5S1_rotx";
+        joint_state.name[1] ="jL5S1_roty";
+        joint_state.name[2] ="jL4L3_rotx";
```

Comment:
Is it necessary to set the joint names each time inside the `while` loop?
Can they be set once and for all before the `while` loop starts and then update only che part that actually changes inside the loop?